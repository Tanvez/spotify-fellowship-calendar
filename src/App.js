import React, { Component } from 'react';
import './App.css';
import { connect } from 'react-redux';
import {receivedEvents, getEvents} from './store'
import {SingleDayModal} from './components/index'

class App extends Component {
  state = { show: false };

  showModal = () => {
    this.setState({ show: true });
  };

  hideModal = () => {
    this.setState({ show: false });
  };
  componentDidMount(){
    this.props.loadInitialData()
  }
  render() {
    const {currentMonth, currentYear, months} = this.props
    let dayArray = [];
    let events = months[currentMonth]
    
    for(let i = 1; i<=31; i++){
      let descript = ''
      let eventsArr = []
      
      if(events[i]){
        for(let evtIdx = 0; evtIdx < events[i].length; evtIdx++){
          descript = events[i][evtIdx].description
          eventsArr.push(<div key = {evtIdx}>{descript}</div>)
        }
      }
      
      dayArray.push(
        <div key={i}>
          <SingleDayModal show={this.state.show} handleClose={this.hideModal}>
            TODO ADD SINGLE DAY EVENT VIEW
          </SingleDayModal>
          <div className='grid-item' onClick={this.showModal} > 
            {i} 
            {
              eventsArr.map(evt=>{
                return evt
              })
            }
          </div>
        </div>
      )
    }
    
    return (
      <div>
      <h1> {currentMonth+' '+currentYear}</h1>
      <div className="grid-container">
        {dayArray.map(listItem=>{
          return listItem
        })
      }
      </div>
      </div>
    );
  }
}

const mapDispatch = (dispatch) => {
  return{
    loadInitialData(){
     dispatch(receivedEvents())
     getEvents()
    }
  }
}

const mapState = (state) => {
  return {
    currentMonth: state.currentMonth,
    currentYear: state.currentYear,
    months: state.months
  }
}
export default connect(mapState, mapDispatch)(App) 
