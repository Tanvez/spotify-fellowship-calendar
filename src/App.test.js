import React from 'react';
import Enzyme, {shallow} from 'enzyme'
import expect from 'expect'
import App from './App';
import configureStore from 'redux-mock-store'
import ReactSixteenAdapter from 'enzyme-adapter-react-16'
Enzyme.configure({adapter: new ReactSixteenAdapter()})

const initialState =
[
  {
    'createdAt':"2018-06-27T20:23:52.691Z",
    'description':"Running test",
    'end':"2018-01-01T16:11:00.000Z",
    'id':11,
    'start':"2018-01-01T16:11:00.000Z",
    'updatedAt':"2018-06-27T20:23:52.691Z",
    'userId':1
  }
]

const mockStore = configureStore()
const store = mockStore(initialState)
const wrapper = shallow(<App store={store} />)
describe('Testing root App.js', ()=>{
    it('renders without crashing', () => {
      expect(wrapper.find('App')).toHaveLength(1)
    })

    it('Capturing Snapshot of App', () => {
      expect(wrapper).toMatchSnapshot()
  })
  }
)
