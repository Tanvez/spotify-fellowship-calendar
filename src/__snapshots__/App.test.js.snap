// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing root App.js Capturing Snapshot of App 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(App)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "events": Array [
        Object {
          "createdAt": "2018-06-27T20:23:52.691Z",
          "description": "Running test",
          "end": "2018-01-01T16:11:00.000Z",
          "id": 11,
          "start": "2018-01-01T16:11:00.000Z",
          "updatedAt": "2018-06-27T20:23:52.691Z",
          "userId": 1,
        },
      ],
      "loadInitialData": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "events": Array [
          Object {
            "createdAt": "2018-06-27T20:23:52.691Z",
            "description": "Running test",
            "end": "2018-01-01T16:11:00.000Z",
            "id": 11,
            "start": "2018-01-01T16:11:00.000Z",
            "updatedAt": "2018-06-27T20:23:52.691Z",
            "userId": 1,
          },
        ],
        "loadInitialData": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
